### ISSUE TITLE
OpenFlow table resets after restart

### ISSUE THREAD

**Issue by emad-nk** [vthapar/odl-openstack#1]
Hi,

Thanks for your devstack config files. I have used all in one config file.
If I install openstack then everything is fine, and I can ping hosts (I use nova-docker). 
However if I restart then I cannot ping hosts anymore and OpenFlow table resets without any entry. 

Do you know what's wrong? Or maybe I should never restart the pc?
And How can I modify the flow through karaf? I use Helium SR4, and when I login into dlux web interface, under add flow section the "node" is empty and I cannot use it.

Thanks in advance

**Comment by vthapar** [comment]
Hi,

Good to hear from you. By restart PC mean the one where you're running
Openstack VMs, right? If yes, then yes, everything will be gone and will
have to run devstack again. Devstack is mainly used for running openstack
off latest code and is a easy starting point. If you want things to persist
and work on restart, you will need to install openstack as services.

What this means is you will need to manually install all the services using
apt-get and configure them. You can refer to your devstack installation for
what entries you will need in your different conf files. I think I also
have some scripts on my github that will let you do the same.

There isn't any way to modify flow from Karaf. I'd recommend using Rest
interface. I am a bit hay on Helium, but think it should be available on
He-SR4. You can lookup ask.opendaylight.org, it is a commonly asked
question so should be able to find answer there. Let me know if unable to,
will have to dig my old notes to find something.

Regards,
Vishal.

**Comment by emad-nk** [comment]
Thank you very much for your quick response. 
I really appreciate it. 

Regards,
Emad

**Comment by emad-nk** [comment]
Hi Vishal,

Sorry to disturb you again. But I was wondering if you could help me with something that I couldn't solve since yesterday. I try to add flow through rest api. 
My openstack instance ip address is: 10.0.0.4
So I want to add a flow for example to stop pinging that ip, and here is the xml:
![screenshot from 2015-11-04 15 34 32](https://cloud.githubusercontent.com/assets/4420745/10939378/d6752caa-8309-11e5-9804-2b864259d461.png)

I get the http response 200, and when I check I can see this flow is added, however I am still able to ping 10.0.0.4. I am not sure if there is a bug in helium, or I am doing something wrong.

**Comment by vthapar** [comment]
Hi,

I'm travelling so responses will be delayed. Looking at your flow, it looks
okay, but can you share flows on your switch? I suspect you have some other
flow that allows traffic and is higher priority than this one. Alternately,
you can increase priority of this flow and see if it works.

Regards,
Vishal.

**Comment by emad-nk** [comment]
Yeah on Friday, I found out that my priority should be increased, so I increased it however still does not work. Here is the flow list, and the first rule that you see is the one that I added.

![screenshot from 2015-11-09 10 08 27](https://cloud.githubusercontent.com/assets/4420745/11028770/253d22de-86ca-11e5-9b75-f1b57729bd83.png)

**Comment by vthapar** [comment]
Default priority for OVS when not specified is 32768, so 18000, is still
lower. Increase it to 33000 and then see if it works.
http://manpages.ubuntu.com/manpages/natty/man8/ovs-ofctl.8.html [search for
priority=value for relevant text].

**Comment by emad-nk** [comment]
yes it worked! Thank you so much! 
So now that ODL integrated with OpenStack, do you know how can I chain OpenStack instances together, so the traffic go through instance1 then 2 then 3?! It seems ODL SFC is not working with OpenStack now.
I want to do it in static way. Any hint would be appreciated

**Comment by vthapar** [comment]
Depends on what sort of practical knowledge you're looking for. If it is
deployment, I am not sure as am more on developer side of things. If you
mean developer/code perspective, I'd recommend involved in OVSDB Project in
OpenDaylight.

**Comment by emad-nk** [comment]
Sorry I edited my previous comment, I want to chain OpenStack instances with ODL for some monitoring measurements. However, I need some hints where to go or what to look for.
